File: backend/app/schemas/idea.py
"""Idea schemas for request/response validation."""
from datetime import datetime
from pydantic import BaseModel, Field


class IdeaCreate(BaseModel):
    """Schema for creating an idea."""

    title: str = Field(..., min_length=1, max_length=200)
    description: str = Field(..., min_length=1)
    priority: str | None = Field(None, pattern="^(low|medium|high|critical)$")


class IdeaUpdate(BaseModel):
    """Schema for updating an idea."""

    title: str | None = Field(None, min_length=1, max_length=200)
    description: str | None = Field(None, min_length=1)
    status: str | None = Field(
        None, pattern="^(backlog|under_review|approved|rejected|implemented)$"
    )
    priority: str | None = Field(None, pattern="^(low|medium|high|critical)$")
    business_value: int | None = Field(None, ge=1, le=10)
    technical_complexity: int | None = Field(None, ge=1, le=10)
    estimated_effort: str | None = Field(None, max_length=100)
    market_fit_analysis: str | None = None
    risk_assessment: str | None = None


class IdeaResponse(BaseModel):
    """Schema for idea response."""

    id: str
    title: str
    description: str
    status: str
    priority: str
    business_value: int | None = None
    technical_complexity: int | None = None
    estimated_effort: str | None = None
    market_fit_analysis: str | None = None
    risk_assessment: str | None = None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class IdeaEvaluationRequest(BaseModel):
    """Schema for requesting AI evaluation of an idea."""

    title: str = Field(..., min_length=1, max_length=200)
    description: str = Field(..., min_length=1)
    context: str | None = Field(None, description="Additional context for evaluation")


class IdeaEvaluationResponse(BaseModel):
    """Schema for AI evaluation response."""

    business_value: int = Field(
        ..., ge=1, le=10, description="Business value score (1-10)"
    )
    technical_complexity: int = Field(
        ..., ge=1, le=10, description="Technical complexity score (1-10)"
    )
    estimated_effort: str = Field(..., description="Estimated effort (e.g., '2 weeks')")
    market_fit_analysis: str = Field(..., description="Market fit analysis")
    risk_assessment: str = Field(..., description="Risk assessment")

File: frontend/src/types/idea.ts
/**
 * Idea management types
 */

export type IdeaStatus = 'backlog' | 'under_review' | 'approved' | 'rejected' | 'implemented'
export type IdeaPriority = 'low' | 'medium' | 'high' | 'critical'

export interface Idea {
  id: string
  title: string
  description: string
  status: IdeaStatus
  priority: IdeaPriority
  business_value: number | null
  technical_complexity: number | null
  estimated_effort: string | null
  market_fit_analysis: string | null
  risk_assessment: string | null
  created_at: string
  updated_at: string
}

export interface IdeaCreate {
  title: string
  description: string
  priority?: IdeaPriority
}

export interface IdeaUpdate {
  title?: string
  description?: string
  status?: IdeaStatus
  priority?: IdeaPriority
  business_value?: number
  technical_complexity?: number
  estimated_effort?: string
  market_fit_analysis?: string
  risk_assessment?: string
}

export interface IdeaEvaluationRequest {
  title: string
  description: string
  context?: string
}

export interface IdeaEvaluationResponse {
  business_value: number
  technical_complexity: number
  estimated_effort: string
  market_fit_analysis: string
  risk_assessment: string
}
